### PR TITLE
Implement scipy connected_components in LCC check

### LINKED ISSUE #112: see if other graph searches are faster than conversion to networkx for LCC check
https://scikit-learn.org/stable/modules/generated/sklearn.utils.graph_shortest_path.graph_shortest_path.html#sklearn.utils.graph_shortest_path.graph_shortest_path

I am guessing the speedup from not switching datatypes completely pays off

### PR DESCRIPTION
**Pull request by kareef928**
#### Reference Issues/PRs

Closes #112 
#### What does this implement/fix? Briefly explain your changes.
Adds scipy connected_components function to the is_fully_connected method (used on scipy csr_matrix and np.ndarray), else uses the original networkx is_connected method (on nx graphs)
#### Any other comments?
Notebook for reference:
https://nbviewer.jupyter.org/github/kareef928/graspologic-personal/blob/main/notebooks/nx-vs-scipy.ipynb

### CODE REVIEW

File: graspologic/utils/utils.py [hunk 1]
```diff
@@ -448,16 +450,23 @@ def is_fully_connected(graph):
...
-            g_object = nx.DiGraph()
-        graph = nx.to_networkx_graph(graph, create_using=g_object)
-    if type(graph) in [nx.Graph, nx.MultiGraph]:
-        return nx.is_connected(graph)
-    elif type(graph) in [nx.DiGraph, nx.MultiDiGraph]:
-        return nx.is_weakly_connected(graph)
+            directed = False
+
+        n_components, labels = connected_components(
+            csgraph=graph, directed=directed, connection="weak", return_labels=True
```
Comment (bdpedigo) [inline graspologic/utils/utils.py:461]: return labels can be false here right?

File: graspologic/utils/utils.py [hunk 2]
```diff
@@ -448,16 +450,23 @@ def is_fully_connected(graph):
     >>> is_fully_connected(a)
     False
     """
+
     if isinstance(graph, (np.ndarray, csr_matrix)):
         if is_symmetric(graph):
-            g_object = nx.Graph()
+            directed = True
```
Comment (asaadeldin11) [inline graspologic/utils/utils.py:456]: shouldn't `directed=False` when graph is symmetric?

File: graspologic/utils/utils.py [hunk 3]
```diff
@@ -448,16 +450,23 @@ def is_fully_connected(graph):
     >>> is_fully_connected(a)
     False
     """
+
     if isinstance(graph, (np.ndarray, csr_matrix)):
         if is_symmetric(graph):
```
Comment (asaadeldin11) [inline graspologic/utils/utils.py:455]:
```suggestion
        directed = not is_symmetric(graph)
```

**Review by asaadeldin11 — APPROVED** [review]
small style change, then this should be all good

File: graspologic/utils/utils.py [hunk 4]
```diff
@@ -448,16 +450,23 @@ def is_fully_connected(graph):
...
-            g_object = nx.Graph()
+            directed = False
         else:
```
Comment (asaadeldin11) [inline graspologic/utils/utils.py:457]:
```suggestion
```

File: graspologic/utils/utils.py [hunk 5]
```diff
@@ -448,16 +450,23 @@ def is_fully_connected(graph):
...
-            g_object = nx.DiGraph()
-        graph = nx.to_networkx_graph(graph, create_using=g_object)
-    if type(graph) in [nx.Graph, nx.MultiGraph]:
-        return nx.is_connected(graph)
-    elif type(graph) in [nx.DiGraph, nx.MultiDiGraph]:
-        return nx.is_weakly_connected(graph)
+            directed = True
```
Comment (asaadeldin11) [inline graspologic/utils/utils.py:458]:
```suggestion
```

**Comment by bdpedigo** [comment]
thanks for the PR @kareef928!
